queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py

```diff
@@ -41,40 +49,71 @@ def process(business: Business,  # pylint: disable=too-many-branches
     """Process the incoming historic conversion filing."""
     # Extract the filing information for incorporation
+    filing_meta.conversion = {}
     if not (conversion_filing := filing.get('filing', {}).get('conversion')):
         raise QueueException(f'CONVL legal_filing:conversion missing from {filing_rec.id}')
+    if business and business.legal_type in ['SP', 'GP']:
+        _process_firms_conversion(business, filing, filing_rec, filing_meta)
+    else:
+        business = _process_corps_conversion(business, conversion_filing, filing, filing_rec)
 
+    return business, filing_rec
+
+
+def _process_corps_conversion(business, conversion_filing, filing, filing_rec):
     if business:
         raise QueueException(f'Business Already Exist: CONVL legal_filing:conversion {filing_rec.id}')
-
     if not (corp_num := filing.get('filing', {}).get('business', {}).get('identifier')):
         raise QueueException(f'conversion {filing_rec.id} missing the business idnetifier.')
-
     # Initial insert of the business record
     business_info_obj = conversion_filing.get('nameRequest')
     if not (business := business_info.update_business_info(corp_num, Business(), business_info_obj, filing_rec)):
         raise QueueException(f'CONVL conversion {filing_rec.id}, Unable to create business.')
-
     if offices := conversion_filing.get('offices'):
         update_offices(business, offices)
-
     if parties := conversion_filing.get('parties'):
         update_parties(business, parties, filing_rec)
-
     if share_structure := conversion_filing.get('shareStructure'):
         shares.update_share_structure(business, share_structure)
-
     if name_translations := conversion_filing.get('nameTranslations'):
         aliases.update_aliases(business, name_translations)
+    return business
 
-    return business, filing_rec
+
+def _process_firms_conversion(business: Business, conversion_filing: Dict, filing_rec: Filing, filing_meta: FilingMeta):
+    # Name change if present
+    with suppress(IndexError, KeyError, TypeError):
+        name_request_json = dpath.util.get(conversion_filing, '/filing/conversion/nameRequest')
+        if name_request_json.get('legalName'):
+            from_legal_name = business.legal_name
+            business_info.set_legal_name(business.identifier, business, name_request_json)
+            if from_legal_name != business.legal_name:
+                filing_meta.conversion = {**filing_meta.conversion, **{'fromLegalName': from_legal_name,
+                                                                       'toLegalName': business.legal_name}}
+    # Update Nature of Business
+    if (naics := conversion_filing.get('filing', {}).get('business', {}).get('naics')) and \
+            (naics_description := naics.get('naicsDescription')):
+        filing_meta.conversion = {**filing_meta.conversion, **{'naicsDescription': naics_description}}
+        business.naics_description = naics_description
```

Comment:
It would be better to update naics using `business_info.update_naics_info` similar to registration.

Reply:
That method has a service call to look up by code. Also it sets key and all that. This one is just a plain text

Reply:
Then can you update naics code as empty/none since its not related.